### PR TITLE
Integer based switch_ip searches in locationlog

### LINKED ISSUE #5792: Searching by 'Source Switch Identifier' for a switch range doesn't work
**Describe the bug**
When you use the switch group search menu on the left in the nodes and search for a switch range, the search is inconsistent with how the data is stored in the DB.

**To Reproduce**
1. Create a switch range 10.0.0.0/24 part of the default switch group
2. Insert the following row in the locationlog (data from a valid RADIUS request):
```
          tenant_id: 1
                mac: 02:00:00:00:00:01
             switch: 10.9.8.7
               port: 0
               vlan: 2
               role: registration
    connection_type: Wireless-802.11-EAP
connection_sub_type: 26
     dot1x_username: administrator
               ssid: 
         start_time: 2020-09-14 08:09:36
           end_time: 0000-00-00 00:00:00
          switch_ip: 10.9.8.7
         switch_mac: NULL
 stripped_user_name: administrator
              realm: null
         session_id: NULL
             ifDesc: NULL
               voip: no
```
3. Insert the following node if necessary
```
          tenant_id: 1
                mac: 02:00:00:00:00:01
                pid: default
        category_id: NULL
        detect_date: 2020-09-14 07:59:40
            regdate: 0000-00-00 00:00:00
          unregdate: 0000-00-00 00:00:00
           lastskip: 0000-00-00 00:00:00
       time_balance: NULL
  bandwidth_balance: NULL
             status: unreg
         user_agent: NULL
       computername: NULL
              notes: NULL
           last_arp: 0000-00-00 00:00:00
          last_dhcp: 0000-00-00 00:00:00
   dhcp_fingerprint: NULL
  dhcp6_fingerprint: NULL
        dhcp_vendor: NULL
   dhcp6_enterprise: NULL
        device_type: NULL
       device_class: NULL
     device_version: NULL
       device_score: NULL
device_manufacturer: NULL
        bypass_vlan: NULL
               voip: no
            autoreg: no
          sessionid: NULL
    machine_account: NULL
     bypass_role_id: NULL
          last_seen: 2020-09-14 08:11:25
```
4. Go in the Nodes, then expand 'Switch Groups' and then expand default and then click on '10.0.0.0/8'
5. It will search for 'Source Switch Identifier' '10.0.0.0/8' and the backend will not expand this value to search for all the switches in that range so our entry will not show up

**Expected behavior**
When searching for a switch range, all the nodes that are connected in a switch in that switch range should show up

**Additional context**
This is on 10.1.0 in the new admin but I tested on 8.1.0 in the old admin and it was also broken.

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Adds a virtual column to the locationlog table allowing integer based searches on nodes using a switch CIDR (10.0.0.0/9 = range 10.0.0.1 - 10.127.255.254).

The switch dropdown in the Nodes section of the webadmin will utilize `greater_than_equal` and `less_than_equal` operators during a ranged node search for CIDR defined switches. Non-CIDR defined switches will continue to work as-is with a single `equals` operator.

# Issue
Fixes #5792 

# Delete branch after merge
YES

# Checklist
- [n/a] Document the feature
- [n/a] Add unit tests
- [n/a] Add acceptance tests (TestLink)

## Enhancements
* Range based node searches from switch CIDR.

## Bug Fixes
* Searching by 'Source Switch Identifier' for a switch range doesn't work (#5792)

### CODE REVIEW

**Comment by satkunas** [comment]
@jrouzierinverse it's all yours.

The 2 `op`s can be seen https://github.com/inverse-inc/packetfence/pull/5820/commits/261749404069b60dffc22758a54b2ec2dd0f7b7e#diff-45c75a1e18b7acf1c565dce953535d3aR87-R88

**Comment by satkunas** [comment]
@julsemaan ready for review

File: html/pfappserver/root/static.alt/src/views/Nodes/index.vue [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default {
               items: switchGroup.members.map(switchGroupMember => {
                 let query
                 if (switchGroupMember.id.indexOf('/') === -1) {
-                  query = { query: JSON.stringify({ op: 'and', values: [{ op: 'or', values: [{ field: 'locationlog.switch', op: 'equals', value: switchGroupMember.id }] }] }) }
+                  query = { query: JSON.stringify({ op: 'and', values: [{ op: 'or', values: [{ field: 'locationlog.switch_ip', op: 'equals', value: switchGroupMember.id }] }] }) }
```
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/views/Nodes/index.vue]: This will not allow search mac based switches

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by julsemaan** [comment]
We need to get this into 10.2 since it will not be possible to backport it to the maintenance

**Comment by julsemaan** [comment]
I'm getting the following error while viewing a node:
```
Sep 24 13:39:00 test-fresh packetfence: pfperl-api(55429) ERROR: Database query failed with non retryable error: The used SELECT statements have a different number of columns (errno: 1222) [SELECT `tenant_id`, `mac`, `switch`, `port`, `vlan`, `role`, `connection_type`, `connection_sub_type`, `dot1x_username`, `ssid`, `start_time`, `end_time`, `switch_ip`, `switch_ip_int`, `switch_mac`, `stripped_user_name`, `realm`, `session_id`, `ifDesc`, `voip` FROM `locationlog` WHERE ( ( `locationlog`.`mac` = ? AND `locationlog`.`tenant_id` = ? ) ) UNION ALL SELECT `tenant_id`, `mac`, `switch`, `port`, `vlan`, `role`, `connection_type`, `connection_sub_type`, `dot1x_username`, `ssid`, `start_time`, `end_time`, `switch_ip`, `switch_mac`, `stripped_user_name`, `realm`, `session_id`, `ifDesc`, `voip` FROM `locationlog_history` WHERE ( ( `locationlog_history`.`mac` = ? AND `locationlog_history`.`tenant_id` = ? ) ) LIMIT ? OFFSET ?]{94:db:c9:38:85:5b, 1, 94:db:c9:38:85:5b, 1, 1001, 0} (pf::dal::db_execute)
```

Guessing because switch_ip_int is not in locationlog_history

**Comment by jrouzierinverse** [comment]
> I'm getting the following error while viewing a node:
> 
> ```
> Sep 24 13:39:00 test-fresh packetfence: pfperl-api(55429) ERROR: Database query failed with non retryable error: The used SELECT statements have a different number of columns (errno: 1222) [SELECT `tenant_id`, `mac`, `switch`, `port`, `vlan`, `role`, `connection_type`, `connection_sub_type`, `dot1x_username`, `ssid`, `start_time`, `end_time`, `switch_ip`, `switch_ip_int`, `switch_mac`, `stripped_user_name`, `realm`, `session_id`, `ifDesc`, `voip` FROM `locationlog` WHERE ( ( `locationlog`.`mac` = ? AND `locationlog`.`tenant_id` = ? ) ) UNION ALL SELECT `tenant_id`, `mac`, `switch`, `port`, `vlan`, `role`, `connection_type`, `connection_sub_type`, `dot1x_username`, `ssid`, `start_time`, `end_time`, `switch_ip`, `switch_mac`, `stripped_user_name`, `realm`, `session_id`, `ifDesc`, `voip` FROM `locationlog_history` WHERE ( ( `locationlog_history`.`mac` = ? AND `locationlog_history`.`tenant_id` = ? ) ) LIMIT ? OFFSET ?]{94:db:c9:38:85:5b, 1, 94:db:c9:38:85:5b, 1, 1001, 0} (pf::dal::db_execute)
> ```
> 
> Guessing because switch_ip_int is not in locationlog_history

Commit 26163f9e7f993c7a42b91372318f34dd4b1d998f fixes this.

**Pull request merged**